cmd/algorand-indexer/main.go

```diff
@@ -94,6 +95,7 @@ func init() {
 
 	rootCmd.PersistentFlags().StringVarP(&postgresAddr, "postgres", "P", "", "connection string for postgres database")
 	rootCmd.PersistentFlags().BoolVarP(&dummyIndexerDb, "dummydb", "n", false, "use dummy indexer db")
+	rootCmd.PersistentFlags().BoolVarP(&dbReadOnly, "db-readonly", "", false, "connect to db in read-only mode")
```

Comment:
I'm not sure this needs to be an option, could it be set implicitly when (a) algod configurations aren't set / found or (b) no-algod is set?

Reply:
Hmm, maybe. It might be valid to run migrations even when there isn't algod? Making --no-algod be read-only does also make sense.